{
  boolean r=false;
  if (exception == null) {
    exception=new IllegalArgumentException("setException(Throwable) was called with a null argument");
  }
  if (!(r=super.setException(exception))) {
    LOG.error("Duplicate exception: " + exception.getMessage());
  }
  return r;
}
